File: Data_Structures/Arrays/threeSum.py
def threeSum(self, nums: List[int] -> List[List[int]]) :
    result = []
    nums.sort()

    for i, a in enumerate(nums):
        if (i>0) and a == nums[i-1]:
            continue
        
        left = i + 1
        right = len(nums) - 1
        while left < right:
            threeSum = a + nums[left] + nums[right]
            if threeSum > 0 :
                right -= 1
            elif threeSum < 0:
                left += 1
            else:
                result.append([a, nums[left], nums[right]])
                left += 1
                #in case there are duplicates and we dont not want any deuplicates answer in our return array
                while nums[left] == nums[left-1] and left < right:
                    left += 1
    return result 

File: Data_Structures/Arrays/threeSum.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var threeSum = function(nums) {
    nums.sort((a, b)=> {
        return a - b});
    let result = [];

    for (let i = 0; i < nums.length; i++) {
        if (i > 0 & nums[i] == nums[i-1]) {
            continue;
        }

        let left = i + 1;
        let right = nums.length-1;

        while (left < right) {
            sum = nums[i] + nums[left] + nums[right];
            if (sum < 0) {
                left += 1;
            } else if (sum > 0) {
                right -= 1;
            } else {
                result.push([nums[i], nums[left], nums[right]]);
                left += 1;

                while (nums[left-1] == nums[left] && left < right) {
                    left += 1;
                }
            }
        }
    }
    return result;
};